src/components/directory-tree-view.js

```diff
@@ -1,31 +1,100 @@
 /**
- * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
  * This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
 import React, { useState } from 'react';
 
+import makeStyles from '@material-ui/core/styles/makeStyles';
 import TreeItem from '@material-ui/lab/TreeItem';
 import TreeView from '@material-ui/lab/TreeView';
 import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
 import ChevronRightIcon from '@material-ui/icons/ChevronRight';
 import { fetchDirectoryContent } from '../utils/rest-api';
 import { useDispatch, useSelector } from 'react-redux';
 import { setCurrentChildren, setSelectedDirectory } from '../redux/actions';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListItemText from '@material-ui/core/ListItemText';
+import AddIcon from '@material-ui/icons/Add';
+import withStyles from '@material-ui/core/styles/withStyles';
+import CreateStudyForm from './create-study-form';
+import { useIntl } from 'react-intl';
+import { elementType } from '../utils/elementType';
+
+const useStyles = makeStyles(() => ({
+    treeItemLabel: {
+        display: 'flex',
+        alignItems: 'center',
+    },
+}));
+
+const StyledMenu = withStyles({
+    paper: {
+        border: '1px solid #d3d4d5',
+    },
+})((props) => (
+    <Menu
+        elevation={0}
+        getContentAnchorEl={null}
+        anchorOrigin={{
+            vertical: 'bottom',
+            horizontal: 'left',
+        }}
+        {...props}
+    />
+));
 
 const DirectoryTreeView = ({ rootDirectory }) => {
+    const classes = useStyles();
+
     const [treeData, setTreeData] = useState(rootDirectory);
     const [expanded, setExpanded] = React.useState([]);
-    const dispatch = useDispatch();
+    const [anchorEl, setAnchorEl] = React.useState(null);
+    const [openAddNewStudyDialog, setOpenAddNewStudyDialog] = React.useState(
+        false
+    );
 
     const selectedDirectory = useSelector((state) => state.selectedDirectory);
 
+    const dispatch = useDispatch();
+    const intl = useIntl();
+
+    const handleOpenMenu = (event) => {
+        setAnchorEl(event.currentTarget);
+        event.stopPropagation();
```

Comment:
why is this needed ?

Reply:
to avoid event propagation the higher order elements